rules/CodeQuality/Rector/FuncCall/SetTypeToCastRector.php

```diff
@@ -98,26 +92,18 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        if (! $this->argsAnalyzer->isArgInstanceInArgsPosition($node->args, 1)) {
-            return null;
-        }
```

Comment:
`$node->isFirstClassCallable()` is needed to be checked early to avoid error:

```
PHP Warning:  Uncaught AssertionError: assert(!$this->isFirstClassCallable()) in php shell code:1
```

to early.

Reply:
Assert is show or not depends on [zend.assertions](https://www.php.net/manual/en/ini.core.php#ini.zend.assertions). setting, which default OFF on our CI.

Reply:
I created PR https://github.com/rectorphp/rector-src/pull/3823 to enable zend.assertions config .